api/pkg/service/catalog/syncer.go

```diff
@@ -165,8 +165,8 @@ func (s *syncer) Process() error {
 
 	setJobState(model.JobRunning)
 
-	catalog := model.Catalog{}
-	db.Model(job).Related(&catalog)
+	catalog := &model.Catalog{}
+	db.Model(job).Association("Catalog").Find(catalog)
```

Comment:
could you please revert to the old style to pass `&catalog` to indicate that the object is modified?